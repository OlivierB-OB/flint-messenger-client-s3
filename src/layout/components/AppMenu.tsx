import AppBar from '@material-ui/core/AppBar';
import Grid from '@material-ui/core/Grid';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Forum from '@material-ui/icons/Forum';
import React, { Fragment } from 'react';
import { MyConversations } from '../../conversations/components/MyConverstions';
import { MyConversationsButton } from '../../conversations/components/MyConversationsButton';
import { MyName } from '../../identity/components/MyName';
import { MyProfileLink } from '../../profileForm/components/MyProfileLink';
import { MyContacts } from '../../users/components/MyContacts';
import { MyContactsButton } from '../../users/components/MyContactsButton';

export function AppMenu() {
  return (
    <Fragment>
      <AppBar position="static" style={{ height: '10vh' }}>
        <Grid container justify="space-between" alignItems="center" style={{ height: '100%' }}>
          <Grid item>
            <Toolbar>
              <Forum fontSize="large" />
              <Typography variant="h3">flint.</Typography>
            </Toolbar>
          </Grid>
          <Grid item>
            <Toolbar>
              <MyName />
            </Toolbar>
          </Grid>
          <Grid item>
            <Toolbar>
              <MyConversationsButton />
              <MyContactsButton />
              <MyProfileLink />
            </Toolbar>
          </Grid>
        </Grid>
      </AppBar>
      <MyContacts />
      <MyConversations />
    </Fragment>
  );
}
