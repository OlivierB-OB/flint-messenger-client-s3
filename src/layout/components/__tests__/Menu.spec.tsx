import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from '../AppMenu';

describe('Menu', () => {
  it('should display the application menu', async () => {
    const component = shallow(<Menu allowNavigation={true} />);
    expect(component).toMatchSnapshot();
  });

  it('should allow disabling the navigation', async () => {
    const component = shallow(<Menu allowNavigation={false} />);
    expect(component).toMatchSnapshot();
  });
});
