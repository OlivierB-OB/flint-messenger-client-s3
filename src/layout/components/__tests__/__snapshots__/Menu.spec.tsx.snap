// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should allow disabling the navigation 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    position="static"
    style={
      Object {
        "height": "10vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="space-between"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Toolbar))>
          <ForumIcon
            fontSize="large"
          />
          <WithStyles(ForwardRef(Typography))
            variant="h3"
          >
            flint.
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Toolbar))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Toolbar))>
          <Connect(MyNameDisplay) />
        </WithStyles(ForwardRef(Toolbar))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>
`;

exports[`Menu should display the application menu 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    position="static"
    style={
      Object {
        "height": "10vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="space-between"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Toolbar))>
          <ForumIcon
            fontSize="large"
          />
          <WithStyles(ForwardRef(Typography))
            variant="h3"
          >
            flint.
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Toolbar))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Toolbar))>
          <Connect(MyNameDisplay) />
        </WithStyles(ForwardRef(Toolbar))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Toolbar))>
          <Connect(ShowConversationsButton) />
          <Connect(ShowContactsButton) />
          <MyProfileLink />
        </WithStyles(ForwardRef(Toolbar))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>
`;
